### DISCUSSION TITLE
async Client API Requests

### DISCUSSION THREAD

**Discussion by 45by90** [LUCIT-Systems-and-Development/unicorn-binance-rest-api#48] [Q&A]
Hello,

Is there a way to make async client API calls, i.e., make a trade call and wait for it while still getting in streaming data?

The websocket stream is blocked while trying to make a trade (maybe a couple of tries).  By the time the trade goes through or fails, the stream rushes to catch up and everything gets screwed up.  Or perhaps something I'm missing?  I've tried things like:

  ```
with c.concurrent.futures.ThreadPoolExecutor() as executor:
      future = executor.submit(self._executor,'BUY',quant,trade_this_price,price_filter)
      trade_r = future.result()
```

But it still blocks the main process (for example, I have a window display updating, which freezes while the order executions are trying to get put through).

Thank you,

JJ

**Comment by oliver-zehentleitner** [comment]
Moved to https://github.com/LUCIT-Systems-and-Development/unicorn-binance-rest-api/issues/49
